app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -0,0 +1,633 @@
+struct LabeledSection<T: CloneRef> {
+    pub label:   text::Area,
+    pub divider: hline::View,
+    pub content: T,
+}
+
+type WideSection = LabeledSection<component_group::wide::View>;
+type ColumnSection = LabeledSection<column_grid::ColumnGrid>;
+
+impl<T: CloneRef> LabeledSection<T> {
+    pub fn new(content: T, app: &Application) -> Self {
+        let logger = Logger::new("LabeledSection");
+        let label = text::Area::new(app);
+        let divider = hline::View::new(logger);
+        Self { label, divider, content }
+    }
+
+    fn set_style(&self, style: &Style) {
+        self.divider.size.set(Vector2(INFINITE, style.section.divider_height));
+        self.label.set_default_color(style.section.heading.color);
+        self.label.set_default_text_size(style.section.heading.size);
+        self.label.set_font(style.section.heading.font.clone());
+        self.label.set_position_y(-0.75 * style.section.heading.size.raw);
+        self.label.set_position_x(3.0 + style.content.padding);
+    }
+}
+
+impl<T: ObjectOps + CloneRef> LabeledSection<T> {
+    fn set_parent(&self, parent: impl ObjectOps) {
+        parent.add_child(&self.content);
+        parent.add_child(&self.label);
+        parent.add_child(&self.divider);
+    }
+}
+
+/// Trait that provides functionality for layouting and layer setting for structs used in the
+/// `LabeledSection`.
+trait SectionContent {
+    fn set_layers(&self, layers: &Layers, scroll_layer: &Layer);
+    fn height(&self) -> f32;
+    fn set_position_top_y(&self, position_y: f32);
+}
+
+impl SectionContent for component_group::wide::View {
+    fn set_layers(&self, layers: &Layers, _scroll_layer: &Layer) {
+        self.model().set_layers(layers);
+    }
+
+    fn height(&self) -> f32 {
+        self.size.value().y
+    }
+
+    fn set_position_top_y(&self, position_y: f32) {
+        self.set_position_y(position_y - self.height() / 2.0);
+    }
+}
+
+impl SectionContent for column_grid::ColumnGrid {
+    fn set_layers(&self, layers: &Layers, scroll_layer: &Layer) {
+        self.model().set_layers(layers, scroll_layer);
+    }
+
+    fn height(&self) -> f32 {
+        self.size.value().y
+    }
+
+    fn set_position_top_y(&self, position_y: f32) {
+        self.set_position_y(position_y);
+    }
+}
+
+impl<T: SectionContent + CloneRef> LabeledSection<T> {
+    fn set_layers(&self, layers: &Layers, scroll_layer: &Layer) {
+        self.content.set_layers(layers, scroll_layer);
+        scroll_layer.add_exclusive(&self.label);
+        self.label.add_to_scene_layer(scroll_layer);
+        scroll_layer.add_exclusive(&self.divider);
+    }
+
+    /// Full height of the section including header.
+    fn height(&self, style: &Style) -> f32 {
+        let label_height = style.section.heading.height();
+        let body_height = self.content.height();
+        let next_section_offset = 17.0; // TODO[MM] this should not be needed.
+        body_height + label_height + next_section_offset
+    }
```

Comment:
Why `T` needs to be `SectionContent` in order to read the `height`? Is this really needed?

Reply:
Because to get the height of the `LabeledSection` we need to know the height of the `SectionContent`, and there is no other trait that allows us to get the size of an arbitrary `DisplayObject` or `ComponentView`. Currently, this information tends to be made available via FRP, which means that the specific instance of the FRP is required to read the `size` output. 

The main alternative to having this here, would be to define some mechanism, for any `ComponentView` to implement a trait (e.g., `SizedView`) that allows us to get its size.